crates/core/src/protocols/evalcheck/subclaims.rs

```diff
@@ -183,9 +207,33 @@ where
 	Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
+pub fn process_composite_sumcheck<U, F>(
+	comp: &CompositeMLE<F>,
+	meta: ProjectedBivariateMeta,
+	eval: F,
+	witness_index: &mut MultilinearExtensionIndex<U, F>,
+	constraint_builders: &mut Vec<ConstraintSetBuilder<F>>,
+	eq_mle: MultilinearExtension<PackedType<U, F>>,
+) -> Result<(), Error>
+where
+	U: UnderlierType + PackScalar<F>,
+	F: TowerField,
+{
+	if !witness_index.has(meta.multiplier_id) {
```

Comment:
Unfortunately, this block will always execute because you add a new multiplier_id for every composite case